src/popup/modules/InitQrCode.js

```diff
@@ -85,3 +85,38 @@ export const initiationProcess = Promise.all([qrCreatorInit, userInterfaceInit])
 }).catch((error) => {
     Logger.logError(error);
 });
+
+/**
+ * Queries Tabs API for 20 times with a delay of 300ms until the tabs have a defined url.
+ * Will reject if url is not defined.
+ * @returns {Promise}
+ */
+async function getsTabWithValidUrl() {
+    let retryCount = 0;
+    const maxRetries = 20;
+    const delay = 300;
+
+    try {
+        return await retryPromise(async () => {
```

Comment:
Hmm, is the `await` here really needed?

Reply:
yes of course, retryPromise returns a promise, it has to wait for it until it's resolved/rejected

Reply:
Promises are there for, so they can run asynchronously and stuff does _not_ have to wait though.
So I'd bet that you do not need the await here. 

All the waiting it only needs should be done by setTimeout and this here can just be returned. Promises are asynchronous – taht's the point, so I guess it should work.

The code does not have to"wait" explicitly to evaluate the Promise result, it "jumps back" to the last execution point, if the result is there. Or am I missing something here?